ledger/participant-local-store/src/test/lib/com/daml/platform/localstore/IdentityProviderConfigStoreTests.scala

```diff
@@ -0,0 +1,307 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.platform.localstore
+
+import com.daml.ledger.api.domain.{IdentityProviderConfig, IdentityProviderId, JwksUrl}
+import com.daml.lf.data.Ref
+import com.daml.logging.LoggingContext
+import com.daml.platform.localstore.api.IdentityProviderConfigStore.{
+  IdentityProviderConfigExists,
+  IdentityProviderConfigNotFound,
+  IdentityProviderConfigWithIssuerExists,
+  TooManyIdentityProviderConfigs,
+}
+import com.daml.platform.localstore.api.IdentityProviderConfigUpdate
+import org.scalatest.freespec.AsyncFreeSpec
+
+import java.util.UUID
+import scala.concurrent.Future
+
+trait IdentityProviderConfigStoreTests extends IdentityProviderConfigStoreSpecBase {
+  self: AsyncFreeSpec =>
+  implicit val lc: LoggingContext = LoggingContext.ForTesting
+
+  def config(): IdentityProviderConfig =
+    IdentityProviderConfig(
+      identityProviderId = randomId(),
+      isDeactivated = false,
+      jwksUrl = JwksUrl.assertFromString("http://example.com/jwks.json"),
+      issuer = UUID.randomUUID().toString,
+    )
+
+  def randomId() = {
+    val id = UUID.randomUUID().toString
+    IdentityProviderId.Id(Ref.LedgerString.assertFromString(id))
+  }
+
+  "identity provider config store" - {
+    "allows to create and load unchanged an identity provider config" in {
+      testIt { tested =>
+        val cfg1 = config()
+        for {
+          res1 <- tested.createIdentityProviderConfig(cfg1)
+        } yield {
+          res1 shouldBe Right(cfg1)
+        }
+      }
+    }
+
+    "disallow to create identity provider config with non unique id" in {
+      val id = randomId()
+      testIt { tested =>
+        val cfg1 = config().copy(identityProviderId = id)
+        val cfg2 = config().copy(identityProviderId = id)
+        for {
+          res1 <- tested.createIdentityProviderConfig(cfg1)
+          res2 <- tested.createIdentityProviderConfig(cfg2)
+        } yield {
+          res1 shouldBe Right(cfg1)
+          res2 shouldBe Left(IdentityProviderConfigExists(id))
+        }
+      }
+    }
+
+    "disallow to create identity provider config with non unique issuer" in {
+      testIt { tested =>
+        val cfg1 = config().copy(issuer = "issuer1")
+        val cfg2 = config().copy(issuer = "issuer1")
+        for {
+          res1 <- tested.createIdentityProviderConfig(cfg1)
+          res2 <- tested.createIdentityProviderConfig(cfg2)
+        } yield {
+          res1 shouldBe Right(cfg1)
+          res2 shouldBe Left(IdentityProviderConfigWithIssuerExists("issuer1"))
+        }
+      }
+    }
+
+    s"disallow to create more than 10 configs" in {
+      testIt { tested =>
+        for {
+          res1 <- Future.sequence((1 to 10).map(_ => tested.createIdentityProviderConfig(config())))
```

Comment:
Could we use the `maxIdentityProviderConfigs` here?

Reply:
sure